src/http.cpp

```diff
@@ -329,10 +329,14 @@ int Exiv2::http(Exiv2::Dictionary& request,Exiv2::Dictionary& response,std::stri
 
                 // search for the body
                 for ( size_t b = 0 ; bSearching && b < lengthof(blankLines) ; b++ ) {
-                    if ( strstr(buffer,blankLines[b]) ) {
+                    const char* blankLinePos = strstr(buffer,blankLines[b]);
+                    if ( blankLinePos ) {
                         bSearching = false ;
-                        body   = (int) ( strstr(buffer,blankLines[b]) - buffer ) + strlen(blankLines[b]) ;
-                        status = atoi(strchr(buffer,' ')) ;
+                        body   = blankLinePos - buffer + strlen(blankLines[b]);
+                        const char* firstSpace = strchr(buffer,' ');
+                        if (firstSpace) {
+                            status = atoi(firstSpace);
```

Comment:
Assuming we never find a space, won't this mean that status will be kept as `200` (=OK)?

Reply:
I considered making more drastic changes to this code, but decided it would be better to just fix the security problems with minimal changes. If I was going to refactor this code, it looks to me like `status` and `result` could be merged into a single variable. The code that I added further down changes the value of `status` if no space is found.